### PR TITLE
Refactor MTE-904 [v114] Fail build if swiftlint fails

### PR DESCRIPTION
**Pull request by isabelrios**
We are using a pre-build step to run swiftlint. This is not maintained by Bitrise and after recent changes, if the swiflint finds an error, we see that in the test result but the build is not failed.
To save time and credits, as soon as the swiftlint fails, the build should stop so that the error is fixed.

In order to do that, we are going to try to change the pre-built step with a custom script.

### CODE REVIEW

**Comment by isabelrios** [comment]
Adding a script step to fail the build when there is a swiftlint error is easy, but the advantage of the pre-built step is that we get the github check and the swiftlint report with the issue. So, I will try to create a script step that mimics the pre-built step so that we don't lose the reporting part of the step which is very useful. 
This will take a bit longer than anticipated, but hopefully we will have a solution soon

File: bitrise.yml [hunk 1]
```diff
@@ -426,6 +426,18 @@ workflows:
     - swiftlint-extended@1:
         inputs:
         - linting_path: "$BITRISE_SOURCE_DIR"
+    - deploy-to-bitrise-io@1.9: {}
```
Comment (isabelrios) [inline bitrise.yml:429]: Adding this deploy step allows us to keep the swiflint github check and so the available results when swiftlint fails

File: bitrise.yml [hunk 2]
```diff
@@ -426,6 +426,18 @@ workflows:
...
+    - script@1:
+        inputs:
+        - content: |-
+            set -e
+            echo "$SWIFTLINT_REPORT"
+            if [ ! -z "$SWIFTLINT_REPORT" ] ;then
```
Comment (isabelrios) [inline bitrise.yml:435]: With this script we check if there is a swiftlint error, if that's the case the build fails. When there are no failures the report file is not created.

File: bitrise.yml (hunk 1 above)
Comment (isabelrios) [reply bitrise.yml:429]: See this commit where I forced a failure: https://github.com/mozilla-mobile/firefox-ios/runs/13033663063
And this commit fixing that failure: https://github.com/mozilla-mobile/firefox-ios/runs/13034848959

File: bitrise.yml [hunk 3]
```diff
@@ -162,7 +162,7 @@ workflows:
             set -euxo pipefail
             echo "$BITRISE_ARTIFACT_PATHS"
 
-            derived_data=${BITRISE_ARTIFACT_PATHS#*|}
+            derived_data=${BITRISE_ARTIFACT_PATHS##*|}
```
Comment (jjSDET) [inline bitrise.yml:165]: what does this extra `#` give us here?

File: bitrise.yml (hunk 2 above)
Comment (jjSDET) [inline bitrise.yml:435]: a really useful tool to have is `shellcheck` to help with some best practices like the use of `! -z`. it's best to use `-n STRING` instead. rule: https://www.shellcheck.net/wiki/SC2236

File: bitrise.yml [hunk 4]
```diff
@@ -426,6 +426,18 @@ workflows:
     - swiftlint-extended@1:
         inputs:
         - linting_path: "$BITRISE_SOURCE_DIR"
+    - deploy-to-bitrise-io@1.9: {}
+    - script@1:
+        inputs:
+        - content: |-
+            set -e
```
Comment (jjSDET) [inline bitrise.yml:433]: does bitrise automatically handle inserting `#!/usr/bin/env bash` here? i've always added it, but i'm not sure what bitrise considers best practice here.

**Review by jjSDET — APPROVED** [review]
nit: minor `shell check` rule violation. Otherwise LGTM!

File: bitrise.yml (hunk 3 above)
Comment (isabelrios) [reply bitrise.yml:165]: The BITRISE_ARITIFACT_PATHS has three different files now (it also stores the swiftlint_report), it had only two before. They are separated by `|` , like:
BITRISE_ARITIFACT_PATHS=path/to/file1|path/to/file2|path/to/Derived_data

With one `#` we get the path after the first `|` with two `##` we get the path after the latest `|` which is what we want.

File: bitrise.yml (hunk 4 above)
Comment (isabelrios) [reply bitrise.yml:433]: oh no, it is not inserted automatically. Not sure either what bitrise consider a good practice. 
We should have the same in all scripts step. Let me add a ticket to define a correct format and apply it to all the scripts we have in bitrise.yml

File: bitrise.yml (hunk 2 above)
Comment (isabelrios) [reply bitrise.yml:435]: Let me add this also to the ticket I commented below, so we stablish a common best practices doc and follow the rules for the whole bitrise.yml file

**Pull request merged**